cluster/images/karmada-agent/Dockerfile

```diff
@@ -1,7 +1,20 @@
+FROM golang:1.17-alpine3.15 AS builder
+
+WORKDIR /go/src/github.com/karmada-io/karmada
+COPY . /go/src/github.com/karmada-io/karmada
+
+ARG GO_LDFLAGS
+
+RUN CGO_ENABLED=0 GO111MODULE=off go build \
```

Comment:
Why disable `GO111MODULE` here?

Reply:
forget to fix this, just keep the same as compile using local machine.